java-manta-client/src/main/java/com/joyent/manta/client/crypto/MantaEncryptedObjectInputStream.java

```diff
@@ -368,6 +506,26 @@ public long skip(final long numberOfBytesToSkip) throws IOException {
             throw e;
         }
 
+        skipInitialBytes();
+
+        /* I would really love to actually use the CipherInputStream.skip() method
+         * when we aren't using authenticated encryption or we are using a AEAD cipher,
+         * but the CipherInputStream method is horribly broken in some cases
+         * and doesn't return back the actual number of bytes skipped. */
```

Comment:
This is useful information!  Please reword the tone a bit into a comment we can keep.  How else is `skip` broken?